dora/core/common/src/main/java/alluxio/collections/NonUniqueFieldIndexInTrie.java

```diff
@@ -0,0 +1,198 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.collections;
+
+import alluxio.collections.FieldIndex;
+import alluxio.collections.IndexDefinition;
+
+import org.apache.commons.collections4.Trie;
+import org.apache.commons.collections4.trie.PatriciaTrie;
+
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicBoolean;
+import javax.annotation.concurrent.NotThreadSafe;
+
+/**
+ * A class representing a non-unique index using a PatriciaTrie. A non-unique index is an index
+ * where an index value can map to one or more objects.
+ *
+ * @param <T> type of objects in this index
+ * @param <V> type of the field used for indexing
+ */
+@NotThreadSafe
+public class NonUniqueFieldIndexInTrie<T, V> implements FieldIndex<T, V> {
+  private final IndexDefinition<T, V> mIndexDefinition;
+  private final Trie<String, Set<T>> mIndexTrie;
+
+  /**
+   * Constructs a new {@link NonUniqueFieldIndexInTrie} instance.
+   *
+   * @param indexDefinition definition of index
+   */
+  public NonUniqueFieldIndexInTrie(IndexDefinition<T, V> indexDefinition) {
+    // Wrap the PatriciaTrie with Collections.synchronizedMap to make it thread-safe
+    mIndexTrie = new PatriciaTrie<>();
```

Comment:
What's the default parameters if you use an empty parameter list to create `PatriciaTrie`?